rsconnect_jupyter/static/connect.js

```diff
@@ -551,8 +561,43 @@ define([
             .addClass("active")
             .siblings()
             .removeClass("active");
+
+          updateCheckboxStates();
         });
 
+        function updateCheckboxStates() {
+          var publishingWithSource = (appMode === 'jupyter-static');
+          var includingFiles = $('#include_files').prop('checked');
+
+          var $filesBox = $('#include_files');
+          $filesBox.prop('disabled', !publishingWithSource);
+          if (!publishingWithSource) {
+            $filesBox.prop('checked', false);
+          }
+          $filesBox.parent().toggleClass('rsc-text-light', !publishingWithSource);
+          
+          var canIncludeSubdirs = publishingWithSource && includingFiles;
+          var $subdirsBox = $('#include_subdirs');
+          $subdirsBox.prop('disabled', !canIncludeSubdirs);
+          if (!canIncludeSubdirs) {
+            $subdirsBox.prop('checked', false);
+          }
+          $subdirsBox.parent().toggleClass('rsc-text-light', !canIncludeSubdirs);
+        }
+
+        function bindCheckbox(id) {
+          // save/restore value in server settings
+          var $box = $('#' + id);
+          $box.prop('checked', config.servers[selectedEntryId][id]);
```

Comment:
`entry` is in scope here, no need to double-subscript: https://github.com/rstudio/rsconnect-jupyter/blob/master/rsconnect_jupyter/static/connect.js#L353